llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -0,0 +1,214 @@
+    'max_seq_len',
+    'model',  # converted into models
+    'tokenizer',  # converted into models
+    'save_folder',  # required, but used as load_path
+}
+OPTIONAL_PARAMS_FOR_EVAL = {
+    'dist_timeout',
+    'eval_gauntlet',
+    'fsdp_config',
+    'icl_subset_num_batches',
+    'loggers',
+    'precision',
+    'python_log_level',
+    'seed',
+}
+
+
+def get_run_name(previous_run_name: str, count: int) -> str:
+    *name_without_uuid_suffix, _ = previous_run_name.split('-')
+    name_suffix = ('-'.join(name_without_uuid_suffix))
+
+    if len(name_suffix) > MAX_RUN_NAME_LENGTH:
+        log.warning(
+            f'Training run name {name_suffix} may be too long, truncating to {MAX_RUN_NAME_LENGTH} characters'
+        )
+        name_suffix = name_suffix[:MAX_RUN_NAME_LENGTH]
+
+    return f'eval{count}-{name_suffix}'
+
+
+def get_load_path(save_folder: str,
+                  save_latest_filename: Optional[str] = None) -> str:
+    # TODO: check that the prefix is remote and not a local file (not supported of course)
+
+    if not save_latest_filename:
+        rank = dist.get_global_rank()
+        save_latest_filename = f'latest-rank{rank}.pt'
+
+    return f'{save_folder}/{save_latest_filename}'
+
+
+def get_eval_models_dict(
+    model: Dict[str, Any],
+    tokenizer: Dict[str, Any],
+) -> List[Dict[str, Any]]:
+    name = model.get('name')
+
+    cfg_overrides = model.pop('cfg_overrides', {})
+    for key in cfg_overrides:
+        model[key] = cfg_overrides[key]
+
+    new_model = {'model_name': name, 'model': model}
+
+    if tokenizer:
+        new_model['tokenizer'] = tokenizer
+
+    return [new_model]
+
+
+class AsyncEval(Callback):
+    """Run the eval loop asynchronously as part of a MosaicML platform run.
+
+    Args:
+        interval: Union[str, int, Time]: The interval describing how often eval runs should be
+            launched. If an integer, it will be assumed to be in :attr:`.TimeUnit.EPOCH`.
+            Otherwise, the unit must be either :attr:`.TimeUnit.EPOCH`, :attr:`.TimeUnit.BATCH`,
+            :attr:`.TimeUnit.TOKEN`, or :attr:`.TimeUnit.SAMPLE`.
+        compute: Optional[Union[ComputeConfig, Dict[str, Any]]]: The compute configuration to
+            use for the eval run. If not provided, the same cluster as the current run and a
+            single GPU node will be used.
+    """
+
+    def __init__(
+        self,
+        interval: Union[str, int, Time],
+        compute: Optional[Union[ComputeConfig, Dict[str, Any]]] = None,
+    ):
+        self.check_interval = create_interval_scheduler(interval)
+        self.compute = compute
+        self.count = 0
+        self.last_launch: Optional[Time] = None
+
+        # Run these during init to fail fast in any of the error cases
+        self.current_run = self._get_current_run()
+        self.get_eval_parameters(
+            self.current_run.submitted_config.parameters or {},
```

Comment:
What happens if the current run params is empty?

Reply:
Updated in b58ccf9c33d6d1b3227762fb7a8c1552a350a56d! missing parameters will raise an error on callback initialization